src/main/kotlin/com/avail/tools/options/OptionProcessorFactory.kt

```diff
@@ -57,32 +59,137 @@ import java.util.*
  * @param optionKeyType
  *   The [type][Class] of option keys.
  */
-class OptionProcessorFactory<OptionKeyType : Enum<OptionKeyType>> constructor(
-	private val optionKeyType: Class<OptionKeyType>)
+class OptionProcessorFactory<OptionKeyType : Enum<OptionKeyType>>
+	private constructor (
+		private val optionKeyType: Class<OptionKeyType>)
 {
+	companion object {
+		inline fun <reified O : Enum<O>> create(
+			noinline setup: OptionProcessorFactory<O>.()->Unit
+		): OptionProcessor<O> = privateCreate(O::class.java, setup)
+
+		/**
+		 * A helper factory function.  Unfortunately, Kotlin's visibility rules
+		 * (effectively inherited from the JVM) for things accessed from inline
+		 * function bodies prevents this from being hidden entirely.
+		 */
+		fun <O : Enum<O>> privateCreate(
+			optionKeyType: Class<O>,
+			setup: OptionProcessorFactory<O>.()->Unit
+		): OptionProcessor<O> =
+			with(OptionProcessorFactory(optionKeyType)) {
+				setup()
+				createOptionProcessor()
+			}
+	}
+
 	/** The [set][Set] of all [options][Option].  */
 	private val allOptions = HashSet<Option<OptionKeyType>>()
 
+	/** The rules to check, in order, for valid options and arguments. */
+	private val allRules =
+		mutableListOf<OptionProcessor<OptionKeyType>.()->Unit>()
+
+	/** A helper class to make option bodies more directly expressible. */
+	data class OptionInvocation<OptionKeyType : Enum<OptionKeyType>>(
+		val processor: OptionProcessor<OptionKeyType>,
+		val keyword: String)
+
 	/**
-	 * Configure the [receiver][OptionProcessorFactory].
-	 *
-	 * @param configurator
-	 *   A function that will configure the receiver.
+	 * A helper class to make bodies of options with arguments more directly
+	 * expressible.
 	 */
-	fun configure(
-			configurator: OptionProcessorFactory<OptionKeyType>.() -> Unit) =
-		configurator()
+	data class OptionInvocationWithArgument<OptionKeyType : Enum<OptionKeyType>>
+	(
+		val processor: OptionProcessor<OptionKeyType>,
+		val keyword: String,
+		val argument: String)
 
 	/**
-	 * Add a new [option][Option].
-	 *
-	 * @param option
-	 *   An [option][Option].
+	 * An optional parameter when defining an [option] or [optionWithArgument],
+	 * to indicate how many occurrences of the option are permitted/required.
+	 * For more precise control, the option body can use
+	 * [OptionProcessor.checkEncountered] during processing, or define a [rule].
 	 */
-	fun addOption(option: Option<OptionKeyType>)
-	{
-		assert(!allOptions.contains(option))
-		allOptions.add(option)
+	@Suppress("unused")
```

Comment:
Is this `@Suppress` necessary? The class is clearly used, as are some of the fields (`OPTIONAL`, at least). Are you suppressing the warning for the other fields?

Reply:
Yeah, it was for some of the fields that didn't happen to get used, but are likely to be in subsequent command line programs.  Originally this was an enum, but I figured it might be useful to allow other reusable cardinalities, like utilities that concatenate at least two of something.